src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,643 @@
+    device_c['installationParam']['latitude'], device_c['installationParam'][
+        'longitude'] = 39.0378, -100.4463
+
+    # Set the AntennaGain and EIRP capability.
+    device_a['installationParam']['eirpCapability'] = 30
+    device_b['installationParam']['eirpCapability'] = 47
+    device_a['installationParam']['antennaGain'] = 16
+    device_b['installationParam']['antennaGain'] = 16
+
+    conditionals_b = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    conditionals = [conditionals_b]
+    del device_b['installationParam']
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['measCapability']
+
+    # Create the actual config.
+    devices = [device_a, device_b, device_c]
+    config = {
+        'registrationRequests': devices,
+        'conditionalRegistrationData': conditionals,
+        'palRecords': pal_records,
+        'sasTestHarnessCert': os.path.join('certs', 'sas.cert'),
+        'sasTestHarnessKey': os.path.join('certs', 'sas.key')
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_PCR_1_default_config)
+  def test_WINNF_FT_S_PCR_1(self, config_filename):
+    """Successful Maximum PPA Creation.
+
+    Checks PPA generated by SAS UUT shall be fully contained within the service area.
+    """
+    # Load the Config file
+    config = loadConfig(config_filename)
+
+    # Inject the PAL records.
+    for pal_record in config['palRecords']:
+      self._sas_admin.InjectPalDatabaseRecord(pal_record)
+
+    # Register devices and  check response.
+    cbsd_ids = self.assertRegistered(config['registrationRequests'],
+                                     config['conditionalRegistrationData'])
+
+    # Trigger SAS UUT to create a PPA boundary.
+    pal_ids = [record['palId'] for record in config['palRecords']]
+    ppa_creation_request = {
+        "cbsdIds": cbsd_ids,
+        "palIds": pal_ids
+    }
+
+    # Trigger PPA Creation to SAS UUT.
+    ppa_id = self.triggerPpaCreationAndWaitUntilComplete(ppa_creation_request)
+    logging.debug('ppa_id received from SAS UUT:%s', ppa_id)
+
+    # Notify SAS UUT about SAS Harness and trigger Full Activity Dump and
+    # retrieves the PPA Zone record.
+    uut_ppa_zone_data = self.triggerFadGenerationAndRetrievePpaZone(
+        ssl_cert=config['sasTestHarnessCert'],
+        ssl_key=config['sasTestHarnessKey'])
+
+    # Write SAS UUT PPA to output directory of PCR.1 test.
+    # PPA Zone received from SAS UUT in PCR.1 test will be considered as input
+    # for PCR 3,PCR 6 and PCR 7 tests.
+    ppa_zone_data_file_path = os.path.join('testcases', 'output', 'test_WINNF_FT_S_PCR_1',
```

Comment:
Will this auto-create the directory if it does not exist?

I think you need to include the config filename in this filename so that you don't overwrite the output from config 1 when you execute with config 2.

Reply:
Agreed   Added the check to ensure that directory is not exist then create and then   dump the ppa_zone data.. Config_file_name is now appended with   sas_uut_claimed_ppa_boundary-default.config.json